pandas/core/indexes/base.py

```diff
@@ -695,13 +695,31 @@ def memory_usage(self, deep=False):
     # ops compat
     @deprecate_kwarg(old_arg_name='n', new_arg_name='repeats')
     def repeat(self, repeats, *args, **kwargs):
-        """
-        Repeat elements of an Index. Refer to `numpy.ndarray.repeat`
-        for more information about the `repeats` argument.
+        """Repeat elements of an Index.
+
+        Parameters
+        ----------
+        repeats : int
+            The number of repetitions for each element.
```

Comment:
There is still some discussion on whether we want to document `args` and `kwargs`, but I think we should. If they are ignored, I think it'd be useful for users (and also developers) to know it, and why they are present. If they are being sent to numpy, that's also something worth mentioning.

Reply:
these are for numpy compat (only a small set of methods are like this). see the validate_repeat below.

Reply:
In this case the `*kwargs` are not ignored, but you get an informative error message if you try to pass any additional keyword (also in case of the numpy ones like `axis`). So therefore I would personally not document them.

Reply:
I would agree that mentioned **kwargs are for numpy compat is not a bad idea

Reply:
I thought it was only for the nice error message, but indeed it is also accepting (not raising) the argument when it is the default value (in this case `axis=None`). 
In that case, it's fine for me as well to actually document this